backend/controllers/todoControllers.js

```diff
@@ -0,0 +1,72 @@
+import { getDataBase } from '../db/index.js'
+import { validateValue } from '../helpers.js'
+
+const db = getDataBase()
+
+export const getAllTodos = async (req, res) => {
+  try {
+    const allTodos = await db.findAsync({})
+    res.send(allTodos)
+  } catch (error) {
+    res.status(500).send({error: "Failed to receive tasks"})
+  }
+  
+}
+
+export const addTask = async (req, res) => {
+  const body = req.body
+  
+  try {
+    await validateValue(body)
+    const task = await db.insertAsync({
+      text: body.text,
+      is_checked: body.is_checked,
+    })
+
+    res.send(task)
+  } catch (error) {
+    res.status(400).send({ error: "Missing required fields" })
+  }
+
+}
+
+export const deleteTask = async (req, res) =>  {
+  const taskId = req.params.id
+
+  const existingTask = await db.findOneAsync({ _id: taskId })
+
+  if (!existingTask) {
+    return res.status(404).send({ error: "Task not found" })
+  }
+
+  try {
+    await db.removeAsync({ _id: taskId })
```

Comment:
Here, instead of first finding, then removing (two round-trips), use the `numRemoved` value returned by `remove`: https://github.com/seald/nedb?tab=readme-ov-file#removing-documents

If the document is found AND removed, `numRemoved > 0`. Otherwise, if the the document was not found, `numRemoved === 0`